src/transports/ws/sws.c

```diff
@@ -773,14 +744,21 @@ static int nn_sws_fail_conn (struct nn_sws *self, int code, char *reason)
 
     self->fail_msg_len += payload_len;
 
-    self->instate = NN_SWS_INSTATE_FAILING;
+    self->instate = NN_SWS_INSTATE_CLOSED;
 
-    /*  On the next recv, the connection will be failed. Why defer
-        until the next recv? Semantically, until then, this incoming
-        message has not been interpreted, so it's not until then that
-        it could be failed. This type of pre-processing is necessary
-        to early fail chunked transfers. */
```

Comment:
@wirebirdlabs I don't think this is quite right.  It looked like it was meant to fail chunked transfers *in the face of a deferred* read notification to the user.  That is, we issue a recv response to the user, but we we don't wait for that to occur before issuing the CLOSE to the peer.  (Put another way, it looks like you were attempting to do the fast fail that autobahn wanted, while still waiting until you got all the chunks to issue the response to the app.)

Since we never wait to issue a response, the behavior I've coded here should represent a "fast-failure" like Autobahn wants to see -- that is we close the websocket as soon as we see a failure, without waiting for the full frame to be received.

Note that it appears that many browsers actually don't do the fast fail correctly here, particularly for mangled UTF-8, at least according tot the test results I've seen.

That said, yes, chunked transfers could reasonably happen.  Yes, fast failure is desirable.  No I don't think it is absolutely necessary.  But as I said, I don't think I've broken this aspect of your original behavior.

I hope that all makes sense.